src/tests/eo/suite/eo_test_general.c

```diff
@@ -199,11 +199,27 @@ _eo_signals_cb_added_deled(void *data EINA_UNUSED, const Efl_Event *event)
      fail_if(callback_array->func != _eo_signals_cb_added_deled);
 }
 
-EFL_CALLBACKS_ARRAY_DEFINE(_eo_signals_callbacks,
-{ EV_A_CHANGED, _eo_signals_a_changed_cb },
-{ EV_A_CHANGED, _eo_signals_a_changed_cb2 },
-{ EV_A_CHANGED, _eo_signals_a_changed_never },
-{ EFL_EVENT_DEL, _eo_signals_efl_del_cb });
+// We don't use the EFL_CALLBACKS_ARRAY_DEFINE macro because we
+// we need the callbacks be called in a specific order
+static Efl_Callback_Array_Item *
+_eo_signals_callbacks(void)
+{
+    static Efl_Callback_Array_Item items[] =
+      {
+          { EV_A_CHANGED, _eo_signals_a_changed_cb },
+          { EV_A_CHANGED, _eo_signals_a_changed_cb2 },
+          { EV_A_CHANGED, _eo_signals_a_changed_never },
+          { 0, _eo_signals_efl_del_cb },
+          { 0, 0 },
```

Comment:
What does the first `0` means here?

Also, maybe this change?
```suggestion
          { 0, _eo_signals_efl_del_cb },
          { 0, NULL },
```

Reply:
I am using `{ 0, 0 }` to make things exactly like the macro expansion does.

Reply:
I see, well, I think it is okay then.